app/Filament/Resources/GameResource/RelationManagers/AchievementsRelationManager.php

```diff
@@ -343,9 +344,9 @@ public function table(Table $table): Table
                     DeleteAction::make(),
                 ])
                     ->visible(fn (Achievement $record): bool => $this->canReorderAchievements()
-                        || $user->can('assignMaintainer', Achievement::class)
-                        || $user->can('manage', AchievementGroup::class)
-                        || $user->can('delete', $record)
+                        || $this->canAssignMaintainer()
+                        || $this->canManageAchievementGroups()
+                        || request()->user()->can('delete', $record)
```

Comment:
`$user` is `Auth::user()`, which I believe is the same as `request()->user()`, so why change it?

Reply:
PHPStan was giving me a hassle here, and the only way to fix it while using `Auth::user()` was to make all this much more verbose.